pkgs/development/libraries/science/math/zn_poly/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv
+, fetchurl
+, gmp
+, python2
+}:
+
+stdenv.mkDerivation rec {
+  version = "0.9";
+  pname = "zn_poly";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "http://web.maths.unsw.edu.au/~davidharvey/code/zn_poly/releases/zn_poly-${version}.tar.gz";
+    sha256 = "1kxl25av7i3v68k32hw5bayrfcvmahmqvs97mlh9g238gj4qb851";
+  };
+
+  buildInputs = [
+    gmp
+  ];
+
+  nativeBuildInputs = [
+    python2 # needed by ./configure to create the makefile
+  ];
+
+  # Tuning (either autotuning or with hand-written paramters) is possible
+  # but not implemented here.
+  # It seems buggy anyways (see homepage).
+  buildFlags = "all libzn_poly.so";
```

Comment:
The libzn_poly.so doesn't seem very portable (for instance macOS uses .dylib, Windows uses .dll). Are you sure it's needed?

Reply:
Yes its needed, its not included in `all`.

I just checked and gentoo and the sage package both special-case darwin with `libzn_poly.dylib`. Neither has a special case for windows, although sage supports windows as far as I know.

I'll add the darwin special case, but I cannot test it.

Reply:
Okay here is the portable way to do this I think: (see https://github.com/NixOS/nixpkgs/blob/804285f589ded484b530750dd1ec03b9052bcdac/pkgs/development/python-modules/libusb1/default.nix#L15)

```
buildFlags = "all libzn_poly${stdenv.hostPlatform.extensions.sharedLibrary}";
```

Reply:
Great! I changed that.

It should be `targetPlatform` though, right?